engines/native/system/system_test.go

```diff
@@ -17,21 +19,81 @@ import (
 	"github.com/taskcluster/taskcluster-worker/runtime/ioext"
 )
 
+func hasString(x string, y []string) bool {
+	for _, s := range y {
+		if s == x {
+			return true
+		}
+	}
+
+	return false
+}
+
 // Note: Constants testGroup, testCat, testTrue, testFalse, testPrintDir, and
 //       testSleep should be defined per platform
 
 func TestSystem(t *testing.T) {
+	var err error
+	var u *User
+
+	currentUser, err := user.Current()
+	assert.NoError(t, err)
+
+	if currentUser.Username != "root" {
```

Comment:
This won't work on windows... Hmm, the reason I had the `system` build constraint was so that we would only run the tests when in a proper context. I guess there could also be contexts where tests are running in root, but we're testing something else and don't want to pollute with garbage users. Considering that the tests are likely to mess up your system, I think hiding then behind a build constraint might be sane.

But I see how auto detecting if the test can run is super nice :)

---
I guess we don't have to fix this now... We can just leave it like this, maybe add a TODO saying something about how this has to be refactored with testing on windows. Or add the build constraint back...